### PR TITLE
V27

### PR DESCRIPTION
**Pull request by Jorerlol**
This should be most of the bits for the update to version 27/28 (due to Beta 2) of Frontier's journal.

Please double-check if it is valid, and if I've missed everything - I haven't done anything in TypeScript, mostly copied previous stuff.

### CODE REVIEW

**Comment by Jorerlol** [comment]
...Ah, I've missed some new station-services.

**Comment by Jorerlol** [comment]
OK, services added in. Not sure about what's going on with Livery, but...

File: src/models/carrier-jump.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+ *    {
+ *      "WarType":"election",
+ *      "Status":"pending",
+ *      "Faction1":{ "Name":"Parcae Dynamic PLC", "Stake":"", "WonDays":0 },
+ *      "Faction2": { "Name":"GD 140 & Co", "Stake":"Ricci Prospect", "WonDays":0 }
+ *    }
+ *  ]
+ * }
+ * ```
+ */
```
Comment (chrisbruford) [inline src/models/carrier-jump.ts:67]: looks like you've pasted the wrong example - this seems to be for Location

File: src/models/carrier-jump-request.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ *"BodyID":1 }
+ */
+
+export class Location extends JournalEvent {
+	CarrierID: number;
+	SystemName: string;
+	Body: string;
+	SystemAddress: number;
+	BodyID: number;
+}
```
Comment (chrisbruford) [inline src/models/carrier-jump-request.ts:20]: Looks like you've used the wrong event here.

File: src/models/carrier-bank-transfer.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { JournalEvent } from "./journal-event.model";
+
+/**
+ * Transfer of credits.
```
Comment (chrisbruford) [inline src/models/carrier-bank-transfer.ts:4]:
```suggestion
 * Written when there's a transfer of credits.
```

File: src/models/carrier-crew-services.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { JournalEvent } from "./journal-event.model";
+
+/**
+ * Player does something to the crew.
```
Comment (chrisbruford) [inline src/models/carrier-crew-services.ts:4]:
```suggestion
 * Written when player does something to the crew.
```

File: src/models/carrier-module-pack.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { JournalEvent } from "./journal-event.model";
+
+/**
+ * A carrier module pack is bought.
+ * Listed as the same as CarrierShipPack.
+ *{ "timestamp":"2020-03-16T09:25:39Z",
+ *"event":"CarrierShipPack",
```
Comment (chrisbruford) [inline src/models/carrier-module-pack.ts:7]:
```suggestion
 *"event":"CarrierModulePack",
```

File: src/models/carrier-stats.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
+import { JournalEvent } from "./journal-event.model";
+
+/**
+ *
```
Comment (chrisbruford) [inline src/models/carrier-stats.ts:4]:
```suggestion
 * Written when owner opens carrier management
 *
```

File: src/models/carrier-trade-order.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { JournalEvent } from "./journal-event.model";
+
+/**
+ * The carrier owner requests the carrier buy or sell goods.
```
Comment (chrisbruford) [inline src/models/carrier-trade-order.ts:4]:
```suggestion
 * Written when the carrier owner requests the carrier buy or sell goods.
```

**Comment by chrisbruford** [comment]
Thanks for putting this together @Jorerlol 👍

File: src/models/carrier-jump-request.ts (hunk 2 above)
Comment (Jorerlol) [reply src/models/carrier-jump-request.ts:20]: Oops. Give me a second.

**Comment by Jorerlol** [comment]
Updated to review - GH is still complaining, but done.

File: src/models/carrier-jump.ts [hunk 8]
```diff
@@ -0,0 +1,93 @@
+import { JournalEvent } from "./journal-event.model";
+import { BodyTypes } from "../enums/body-types.enum";
+import { Faction } from "./faction.model";
+import { Conflict } from "./conflict";
+
+/**
+ * When written: on startup or resurrection
```
Comment (chrisbruford) [inline src/models/carrier-jump.ts:7]:
```suggestion
 * This is similar to FSDJump and Location, but it is written if the player is online and docked at a fleet carrier when the carrier jumps. Note it does not include the distance jumped, or fuel used.
```

**Pull request merged**